app/src/main/java/com/matryoshka/projectx/data/cache/SharedPrefsCache.kt

```diff
@@ -0,0 +1,70 @@
+package com.matryoshka.projectx.data.cache
+
+import android.content.Context
+import android.util.Log
+import com.google.gson.Gson
+import com.matryoshka.projectx.utils.sharedPreferences
+
+private const val TAG = "SharedPrefsCache"
+
+class SharedPrefsCache<T>(
+    private val objectClass: Class<T>,
+    private val key: String,
+    private val expiredPeriod: Long,
```

Comment:
```suggestion
    private val lifeSpan: Duration
```
Если фундаментально, то посмотри java.time пакет:
https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/TimeUnit.html
https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html
Но так как мы пишем на Kotlin, можно использовать:
https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.time/-duration/

`val lifeSpan = 30.seconds`

Reply:
Исправил